### PR TITLE
Make sure simulated events don't warn when providing extra event properties

### LINKED ISSUE #6379: ReactTestUtils.Simulate logs unexpected console error
To reproduce, open http://jsfiddle.net/0jjnaghw/
Error is logged to browser dev tools (in Chrome & Firefox, somehow not in Safari).

Happens on v15.0.0-rc.2 and on latest master version.

The assignment of additional event data to the newly created SyntheticEvent triggers a console error:
https://github.com/facebook/react/blob/3a4e1dbb5b2957875cab6af11a5e5934e125a0d8/src/test/ReactTestUtils.js#L502

> "Warning: This synthetic event is reused for performance reasons."

### PR DESCRIPTION
**Pull request by zpao**
fixes #6379

### CODE REVIEW

**Comment by zpao** [comment]
I think in an ideal world we'd be using the right SyntheticEvent subclass constructor for the event. This would allow us to set the properties directly on the `fakeNativeEvent` before passing it to the constructor, then all the appropriate properties would be copied off `fakeNativeEvent` (Eg, a click event would go through SyntheticMouseEvent and look at the MouseEventInterface which includes clientX). This would require more work though so I didn't want to get into it now.

File: src/test/__tests__/ReactTestUtils-test.js [hunk 1]
```diff
@@ -474,6 +474,29 @@ describe('ReactTestUtils', function() {
     expect(handler).not.toHaveBeenCalled();
   });
 
+  it.only('should not warn when simulating events with extra properties', function() {
```
Comment (gaearon) [inline src/test/__tests__/ReactTestUtils-test.js]: .only

**Comment by gaearon** [comment]
Re: ideal world, relevant to #6100.

**Comment by zpao** [comment]
Hmm, worth noting that this silences an expected warning (easier to see now on travis now that other issues are cleaned up). I'll look into that some more…

**Comment by sebmarkbage** [comment]
Is the travis test error relevant? Looks plausible.

**Comment by zpao** [comment]
Nope, that's because of #6228. Will take a closer look at that test (only failing in non-createElement mode) in the morning but this should be fine.

**Comment by jimfb** [comment]
:+1:

**Pull request merged**